File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0;
        int right = 0;
        int minLen = Integer.MAX_VALUE;
        int currSum = 0;

        for(; right<nums.length; right++) {
            currSum = currSum + nums[right];
            while (currSum >= target) {
                int len = (right - left) + 1;

                minLen = Math.min(minLen, len);
                currSum -= nums[left];
                left++;
            }
            
        }

        if(minLen == Integer.MAX_VALUE)
            return 0;
        else return minLen;
    }
}

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.cs
public class Solution {
    public int MinSubArrayLen(int target, int[] nums) {
        if(nums[0] == target) return 1;
        int left =0;
        int right = 0;
        int sum = 0;
        int minLen = Int32.MaxValue;
        while(right < nums.Length) {
            sum += nums[right];
            
            while(sum >= target) {
                minLen = Math.Min(minLen, (right - left) + 1);
                sum -= nums[left];
                left++;
            }

            right++;
        }

        if(minLen == Int32.MaxValue) return 0;

        return minLen;
    }
}